### ISSUE TITLE
Indentation in fallthrough CASE that contains a closure

### ISSUE THREAD

**Issue by fixpunkt** [squizlabs/PHP_CodeSniffer#1478]
Given this example of a fallthrough case:

```PHP
<?php
namespace Test;

class Test
{
    public function m($a)
    {
        switch ($a) {
            case 0:
                function () {
                };
            case 1:
                break;
        }
    }
}
```

The line containing the second case statement is highlighted with the message `Line indented incorrectly; expected 16 spaces, found 12`. We spent some time debugging and narrowed it down to [this special case handling](https://github.com/squizlabs/PHP_CodeSniffer/blob/6673f85b4a1d2197be19e37d45e173abcf5c5f15/src/Standards/Generic/Sniffs/WhiteSpace/ScopeIndentSniff.php#L984-L1022), which skips the opening curly which would [cause a scope to be created for the closure ](https://github.com/squizlabs/PHP_CodeSniffer/blob/6673f85b4a1d2197be19e37d45e173abcf5c5f15/src/Standards/Generic/Sniffs/WhiteSpace/ScopeIndentSniff.php#L1053). However, the closing curly is processed and causes [the outer scope to be popped](https://github.com/squizlabs/PHP_CodeSniffer/blob/6673f85b4a1d2197be19e37d45e173abcf5c5f15/src/Standards/Generic/Sniffs/WhiteSpace/ScopeIndentSniff.php#L586) by mistake. This, in turn, causes [these checks](https://github.com/squizlabs/PHP_CodeSniffer/blob/6673f85b4a1d2197be19e37d45e173abcf5c5f15/src/Standards/Generic/Sniffs/WhiteSpace/ScopeIndentSniff.php#L738-L739) to fail, leading to the observed error.

I'm not sure how to fix this, so unfortunately cannot provide a PR right now. Any ideas how to fix this without breaking other cases?

**Comment by gsherwood** [comment]
Turns out the open closure wasn't adding the scope an internal stack, but closing the closure was removing from the stack. That stack is what the CASE statements use to figure out if they have shared closers, so it was breaking.
